presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestDecimalSumAggregation.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.LongDecimalWithOverflowState;
+import com.facebook.presto.operator.aggregation.state.LongDecimalWithOverflowStateFactory;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.UnscaledDecimal128Arithmetic;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.math.BigInteger;
+
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.facebook.presto.spi.type.UnscaledDecimal128Arithmetic.unscaledDecimal;
+import static org.testng.Assert.assertEquals;
+
+public class TestDecimalSumAggregation
+{
+    private static final BigInteger TWO = new BigInteger("2");
+    private static final DecimalType TYPE = createDecimalType(38, 0);
+
+    private LongDecimalWithOverflowState state;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        state = new LongDecimalWithOverflowStateFactory().createSingleState();
```

Comment:
Covered by `TestLongAverage/SumAggregation` classes. This is just unit test for arithmetic.